gcalbot/gcalbot/http.go

```diff
@@ -38,15 +47,182 @@ func NewHTTPSrv(
 	}
 	h.OAuthHTTPSrv = base.NewOAuthHTTPSrv(stats, kbc, debugConfig, oauthConfig, h.db, h.handler.HandleAuth,
 		"gcalbot", base.Images["logo"], "/gcalbot")
-	http.HandleFunc("/gcalbot", h.healthCheckHandler)
+	http.HandleFunc("/gcalbot", h.configHandler)
 	http.HandleFunc("/gcalbot/home", h.homeHandler)
 	http.HandleFunc("/gcalbot/image/screenshot", h.screenshotHandler)
 	http.HandleFunc("/gcalbot/events/webhook", h.handleEventUpdateWebhook)
 	return h
 }
 
-func (h *HTTPSrv) healthCheckHandler(w http.ResponseWriter, r *http.Request) {
-	_, _ = fmt.Fprintf(w, "OK")
+var reminders = []ReminderType{
+	{"0", "At time of event"},
+	{"1", "1 minute before"},
+	{"5", "5 minutes before"},
+	{"10", "10 minutes before"},
+	{"15", "15 minutes before"},
+	{"30", "30 minutes before"},
+	{"60", "60 minutes before"},
+}
+
+func (h *HTTPSrv) configHandler(w http.ResponseWriter, r *http.Request) {
+	var err error
+	defer func() {
+		if err != nil {
+			h.Errorf("error in configHandler: %s", err)
+			w.WriteHeader(http.StatusInternalServerError)
+			_, _ = w.Write([]byte("something went wrong :("))
+		}
+	}()
+
+	err = r.ParseForm()
+	if err != nil {
+		return
+	}
+
+	keybaseUsername, keybaseConvID, ok := h.authUser(w, r)
+	if !ok {
+		h.showLoginInstructions(w)
+		return
+	}
+
+	accountNickname := r.Form.Get("account")
+	calendarID := r.Form.Get("calendar")
+
+	previousAccountNickname := r.Form.Get("previous_account")
+	previousCalendarID := r.Form.Get("previous_calendar")
+
+	reminderInput := r.Form.Get("reminder")
+	inviteInput := r.Form.Get("invite")
+
+	accounts, err := h.db.GetAccountNicknameListForUsername(keybaseUsername)
+	if err != nil {
+		return
+	}
+
+	page := ConfigPage{
+		Title:         "gcalbot | config",
+		KeybaseConvID: keybaseConvID,
+		Account:       accountNickname,
+		Accounts:      accounts,
+		Reminders:     reminders,
+	}
+
+	if accountNickname == "" {
+		h.servePage(w, "config", page)
+		return
+	}
+
+	accountID := GetAccountID(keybaseUsername, accountNickname)
+
+	token, err := h.db.GetToken(accountID)
+	if err != nil || token == nil {
+		return
```

Comment:
redirect to login page (so they are told to configure) here?

Reply:
I'm not sure if that makes sense because they have already selected their account. This should be cleared up when I refactor the oauth stuff because "accounts" and "tokens" will be together in the db so there is no need for two lookups